source/features/add-keyboard-shortcuts-to-comment-fields.js

```diff
@@ -40,16 +41,20 @@ export default function () {
 				event.preventDefault();
 			}
 		} else if (event.key === 'Escape') {
-			const inlineCancelButton = select('.js-hide-inline-comment-form', field.form);
+			// Cancel buttons have different classes for inline comments and editable comments
+			const cancelButton = select(`
+				.js-hide-inline-comment-form,
+				.js-comment-cancel-button
+			`, field.form);
 
-			// Cancel comment if inline, blur the field if it's a regular comment
-			if (field.value === '') {
+			// Cancel if there is a button, else blur the field
+			if (cancelButton) {
```

Comment:
Consideration: It's missing `field.value === ''` but I think it wasn't necessary because:

- `esc` here will be exactly like clicking the `Cancel` button
- If the field is not empty, the user should be asked for confirmation with `add-confirmation-to-comment-cancellation`

Reply:
For now, I've found these inline fields to be "edit comment field" and "add inline comment", both of which check if the input is empty before hiding themselves.